src/generators/rust/renderers/UnionRenderer.ts

```diff
@@ -46,7 +47,21 @@ export const RUST_DEFAULT_UNION_PRESET: UnionPresetType<RustOptions> = {
   },
 
   structMacro({ model, renderer }) {
-    return renderer.renderMacro(model);
+    const blocks = [renderer.renderMacro(model)];
+    if (model.originalInput.discriminator !== undefined) {
+      // when discriminator is provided, polymorphic type is internally-tagged: 
+      // https://serde.rs/enum-representations.html#internally-tagged
+      const discriminatorBlock = `#[serde(tag = "${model.originalInput.discriminator}")]`;
+      blocks.push(discriminatorBlock);
```

Comment:
Eventually, we should probably look at how a discriminator keyword can live within the metamodel itself, instead of relying on originalInput.

Reply:
SGTM. I'll stay tuned for that! 👀